### ISSUE TITLE
DueTask scheduled inconsistent 	

### ISSUE THREAD

**Issue by jvancoillie** [Guikingone/SchedulerBundle#105]
There is a problem with the recovery and execution of scheduled tasks. 

task configuration : 
```yaml
scheduler_bundle:
  transport:
    dsn: 'doctrine://default?execution_mode=first_in_first_out'
  tasks:
    bar:
      type: 'command'
      command: 'debug:router'
      description: 'Display current routes for an application'
```
See below
<img width="1452" alt="Capture d’écran 2021-05-09 à 13 20 13" src="https://user-images.githubusercontent.com/21243943/117570279-b82ffd80-b0c9-11eb-9d77-f96952aa9ade.png">

when setting up the tasks we can see that the next execution is set `2021-05-09T11:19:00+00:00`

I then request a consume, the task is executed and the date of last execution `2021-05-09T11:18:39+00:00` is less than the initial date of execution!

**Comment by jvancoillie** [comment]
after search, it is a limit of the [cron-expression](https://github.com/dragonmantank/cron-expression/blob/76ec7333bd29db2fe3c883b311412a12e5022796/src/Cron/CronExpression.php#L298) parser, the `isDue()` method does not take seconds into account. 

```php
/**
 * Determine if the cron is due to run based on the current date or a
 * specific date.  This method assumes that the current number of
 * seconds are irrelevant, and should be called once per minute.
 *
 * @param string|\DateTimeInterface $currentTime Relative calculation date
 * @param null|string               $timeZone    TimeZone to use instead of the system default
 *
 * @return bool Returns TRUE if the cron is due to run or FALSE if not
 */
```

So, when I run several `scheduler:consume` in the same minute the task is executed each time.

>(and should be called once per minute)

Is it a problem ? do we need to add a check for this ?

**Comment by Guikingone** [comment]
Hum, currently and apart from the `scheduler:consume` command, I don't know how to bypass this limitation, maybe we can pause the worker (thanks to `sleep`) until the next minute but the issue is located in the scheduler, not the worker. 

Regarding the `getDueTasks()` method, I don't know how to add a check in it, maybe via a flag (maybe `hasBeenCalled`?) or an attribute but for me, the issue is related to the external library, not the bundle 🙁

**Comment by jvancoillie** [comment]
yes i agree with you i don't see how to improve that. you can close this request if you wish

**Comment by Guikingone** [comment]
Actually, there might be a solution (note that I use "might" and not "there's a solution" 😄): 

- When retrieving tasks using `getDueTasks()`, we can filter the results using `last_execution` option, this one is set using a `DatetimeImmutable` each time that the task is executed. 

Maybe that we can do something like this: 

```bash
Given the current date (using a specific format)
When I retrieve the due tasks 
And that I filter each one using CronExpression()->isDue()
And that I check if the current date (close to the current minute) does not equal the last execution of the task
Then I should find a set of tasks (or none if tasks have already been executed
```

WDYT?

**Comment by Guikingone** [comment]
Hi @jvancoillie 

I'm gonna merge the fix this morning, should be released in `0.4.5` 🙂

**Comment by jvancoillie** [comment]
great new ! will test this soon 😉
